### ISSUE TITLE
Document soft deleted entities restoration restrictions in The Things Stack

### ISSUE THREAD

**Issue by ymgupta** [TheThingsIndustries/lorawan-stack-docs#1165]
#### Summary

In The Things Stack, once entities are soft deleted, they will move to the **Deleted (Admin)** section within The Things Stack Console. Restoring the entity's access is possible with the admin user account only. If the user hasn't attempted to restore the entities within the 24-hour period, the entity cannot be restored.

#### Why do we need this ?

To assist users in restoring the accidentally deleted entities in The Things Stack.

#### What is already there? What do you see now?

[Purging Entities](https://www.thethingsindustries.com/docs/the-things-stack/management/purge/)

#### What is missing? What do you want to see?

Add the Information about restoring soft deleted entities in The Things Stack.
#### How do you propose to document this?

**Insights:** 

- Entities restoration can be possible with the admin user account only. If the users don't have admin access they need to reach out to the admin user before the default restoration window (i.e., 24 hours) to restore the deleted entity. If an entity is attempted to be restored after 24 hours of deletion, it cannot be restored.
- Whereas in The Things Stack Enterprise deployments, users can configure the required Entities restoration period. 
Ref: [Identity Server Options](https://www.thethingsindustries.com/docs/reference/configuration/identity-server/#general-options)
 
#### Can you do this yourself and submit a Pull Request?

No, @nejraselimovic

**Comment by nejraselimovic** [comment]
Is it possible to soft delete and restore end devices? I don't see this panel for devices

**Comment by nejraselimovic** [comment]
Also, is it possible to see soft-deleted apps and gateways? Like for users we have:

`ttn-lw-cli user list --deleted --all`

If I replace `users` with `application`for example, I get this:

`error:pkg/identityserver/store:account_not_found (user account with id `snejra` not found)`

Not sure if I'm trying something that's not really possible.. @nicholaspcr

**Comment by nicholaspcr** [comment]
> Is it possible to soft delete and restore end devices? I don't see this panel for devices

It is **not** possible to restore deleted end-devices, as those are not soft-deleted but permanently deleted.


> Also, is it possible to see soft-deleted apps and gateways? Like for users we have:
> 
> `ttn-lw-cli user list --deleted --all`
> 
> If I replace `users` with `application`for example, I get this:
> 
> `error:pkg/identityserver/store:account_not_found (user account with id `snejra` not found)`
> 
> Not sure if I'm trying something that's not really possible.. @nicholaspcr

It should be possible and what you described is a bug (thanks for finding it), I did a bit of debugging and will do a fix for it that should be available in `v3.28.0`. Probably worth adding the usage of it in the documentation.

While the fix is still in progress the expected output of the command with the `--deleted` flag has an additional field which is `deleted_at`.

Example:
```
go run ./cmd/ttn-lw-cli app list --deleted
INFO    Telemetry is enabled. Check the documentation for more information on what is collected and how to disable it   {"documentation_url": "https://www.thethingsindustries.com/docs/reference/telemetry/cli"}
INFO    Use the flags "--limit=0 --page=1" to get the next page of results      {"total": 1}
[{
  "ids": {
    "application_id": "app-1"
  },
  "created_at": "2023-09-25T10:59:42.150922Z",
  "updated_at": "2023-09-25T10:59:42.150922Z",
  "deleted_at": "2023-09-25T10:59:51.776821Z"
}]
```

**Comment by KrishnaIyer** [comment]
> Restoring the entity's access is possible with the admin user account only. If the user hasn't attempted to restore the entities within the 24-hour period, the entity cannot be restored.

@ymgupta: Where are you seeing this? I've never heard of this. Indeed as @nicholaspcr said, once an entity is deleted, it's gone. We only have the option to retain and separately purge the IDs. That's a security counter-measure in order to allow a certain time to elapse after deletion of an ID before another user can register it.
